crates/rattler_conda_types/src/repo_data_record.rs

```diff
@@ -0,0 +1,26 @@
+//! Defines the `[RepoDataRecord]` struct.
+
+use crate::PackageRecord;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+/// Information about a package from repodata. It includes a [`crate::PackageRecord`] but it also stores
+/// the source of the data (like the url and the channel).
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Ord, PartialOrd, Clone)]
+pub struct RepoDataRecord {
+    /// The data stored in the repodata.json.
+    #[serde(flatten)]
+    pub package_record: PackageRecord,
+
+    /// The filename of the package
+    #[serde(rename = "fn")]
+    pub filename: String,
```

Comment:
Rust uses `Path::file_name()` - I wonder if we should also use a `_` between `file` and `name`? (same for `file_size`, etc.

Reply:
Good one! Will change!